worlds/noita/__init__.py

```diff
@@ -36,21 +39,21 @@ class NoitaWorld(World):
     web = NoitaWeb()
 
     # Returned items will be sent over to the client
-    def fill_slot_data(self):
-        return {name: getattr(self.multiworld, name)[self.player].value for name in self.option_definitions}
+    def fill_slot_data(self) -> Dict[str, Any]:
+        return self.options.as_dict("death_link", "victory_condition", "path_option", "hidden_chests",
+                                    "pedestal_checks", "orbs_as_checks", "bosses_as_checks", "extra_orbs", "shop_price")
 
     def create_regions(self) -> None:
-        Regions.create_all_regions_and_connections(self.multiworld, self.player)
-        Events.create_all_events(self.multiworld, self.player)
+        Regions.create_all_regions_and_connections(self)
 
     def create_item(self, name: str) -> Item:
         return Items.create_item(self.player, name)
 
     def create_items(self) -> None:
```

Comment:
I wouldn't deny based on this, but I really don't like all these nested calls just for the sake of it. it's hard to follow, and every function created takes up more memory and time.